pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -188,15 +204,18 @@ func (pc *PodCache) notifyWorkloadHandlers(pod *v1.Pod, ev model.Event) {
 		return
 	}
 	// fire instance handles for workload
-	ep := NewEndpointBuilder(pc.c, pod).buildIstioEndpoint(pod.Status.PodIP, 0, "", model.AlwaysDiscoverable, model.Healthy)
-	workloadInstance := &model.WorkloadInstance{
-		Name:      pod.Name,
-		Namespace: pod.Namespace,
-		Kind:      model.PodKind,
-		Endpoint:  ep,
-		PortMap:   getPortMap(pod),
+	podIPs := GetPodIPs(pod)
+	for _, itemIP := range podIPs {
+		ep := NewEndpointBuilder(pc.c, pod).buildIstioEndpoint(itemIP, 0, "", model.AlwaysDiscoverable, model.Healthy)
+		workloadInstance := &model.WorkloadInstance{
+			Name:      pod.Name,
+			Namespace: pod.Namespace,
+			Kind:      model.PodKind,
+			Endpoint:  ep,
+			PortMap:   getPortMap(pod),
+		}
+		pc.c.handlers.NotifyWorkloadHandlers(workloadInstance, ev)
```

Comment:
Workload handler keys by name+namespace: https://github.com/istio/istio/blob/0969ae43cfca4b13b8f1c9c82f6137cbe31ad82a/pilot/pkg/serviceregistry/serviceentry/controller.go#L468. This is now no longer unique. This will need further work to support multiple IP. Currently this will be broken, only last IP used.

Reply:
Hi @howardjohn, yeah, it's not unique for `WorkloadInstance` with multiple IPs. Therefore, there are 2 direction for this:
1. We can change the `configKey` or even `model.ConfigKey` to make sure that each config key can represent only one `WorkloadInstance`
2. We also can change the `WorkloadInstance` to support that one `WorkloadInstance` contains one `IstioEndpoint` which can holds multiple endpoint addresses [here](https://github.com/istio/istio/blob/a8f2c0a3b89297a0761cfd8c77488ebe43a6c007/pilot/pkg/model/service.go#L445)

What do you thinking about this 2 directions, or do you think which one should be reasonable?

Reply:
Hi @howardjohn, the first choice should be the least impact to Istio.  I will change it in another PR, this may split the task into smaller pieces.

Reply:
Hi @howardjohn, I submitted a PR to fix this problem, make the service instance to be unique identify here: https://github.com/istio/istio/pull/45321.